queue_services/entity-emailer/tests/unit/email_processors/test_restoration_notification.py

```diff
@@ -0,0 +1,135 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The Unit Tests for the Restoration email processor."""
+
+from unittest.mock import patch
+
+import requests_mock
+import base64
+
+from entity_emailer.email_processors import restoration_notification
+from tests.unit import prep_restoration_filing
+
+
+def test_complete_full_restoration_notification_includes_notice_of_articles_and_incorporation_cert(session, config):
+    """Test completed full restoration notification."""
+    # setup filing + business for email
+    status = 'COMPLETED'
+    legal_name = 'test business'
+    business_id = 'BC1234567'
+    token = 'token'
+    filing = prep_restoration_filing(session, business_id, '1', status, 'BC', legal_name)
+    with requests_mock.Mocker() as m:
+        m.get(f'{config.get("LEGAL_API_URL")}/businesses/{business_id}/filings/{filing.id}?type=noticeOfArticles',
+              content=b'pdf_content_1', status_code=200)
+        m.get(f'{config.get("LEGAL_API_URL")}/businesses/{business_id}/filings/{filing.id}?type=certificateOfRestoration',
+              content=b'pdf_content_2')
+        output = restoration_notification.process({
+            'filingId': filing.id,
+            'type': 'restoration',
+            'option': status
+        }, token)
+        assert 'content' in output
+        assert 'attachments' in output['content']
+        assert len(output['content']['attachments']) == 2
+        assert output['content']['attachments'][0]['fileName'] == 'Notice of Articles.pdf'
+        assert base64.b64decode(output['content']['attachments'][0]['fileBytes']).decode('utf-8') == 'pdf_content_1'
+        assert output['content']['attachments'][1]['fileName'] == 'Certificate of Restoration.pdf'
+        assert base64.b64decode(output['content']['attachments'][1]['fileBytes']).decode('utf-8') == 'pdf_content_2'
+
+
+def test_paid_restoration_notification_includes_receipt_and_restoration_application_attachments(session, config):
+    """Test PAID full restoration notification."""
+    # setup filing + business for email
+    legal_name = 'test business'
+    business_id = 'BC1234567'
+    status = 'PAID'
+    filing = prep_restoration_filing(session, business_id, '1', status, 'BC', legal_name)
+    token = 'token'
+    with requests_mock.Mocker() as m:
+        m.post(f'{config.get("PAY_API_URL")}/{filing.payment_token}/receipts',
+               content=b'pdf_content_1', status_code=201)
+        m.get(f'{config.get("LEGAL_API_URL")}/businesses/{business_id}/filings/{filing.id}',
+              content=b'pdf_content_2', status_code=200)
```

Comment:
Paid restorations triggers the sending of the receipt and the restoration application